File: 1007_Minimum_Domino_Rotations_For_Equal_Row.cpp
class Solution {
public:
    int minDominoRotations(vector<int>& p, vector<int>& q) {
        for (int z : {p[0], q[0]}) {
            int i = 0, x = 0, y = 0;
            for (; i < p.size(); ++i) {
                if (p[i] != z && q[i] != z) break;
                x += p[i] != z;
                y += q[i] != z;
            }
            if (i == p.size()) return min(x, y);
        }
        return -1;
    }
};

File: 1007_Minimum_Domino_Rotations_For_Equal_Row.py
class Solution(object):
    def minDominoRotations(self, p, q):
        for d in (p[0], q[0]):
            a = b = 0
            for i in range(len(p)):
                if p[i] != d and q[i] != d: break
                if p[i] != d: a += 1
                if q[i] != d: b += 1
            else:
                return min(a, b)
        return -1
